x/autopilot/keeper/redeem_stake_test.go

```diff
@@ -0,0 +1,253 @@
+package keeper_test
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/cometbft/cometbft/crypto/ed25519"
+	"github.com/cosmos/ibc-go/v7/modules/apps/transfer"
+	transfertypes "github.com/cosmos/ibc-go/v7/modules/apps/transfer/types"
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v7/modules/core/04-channel/types"
+
+	recordsmodule "github.com/Stride-Labs/stride/v16/x/records"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	router "github.com/Stride-Labs/stride/v16/x/autopilot"
+	"github.com/Stride-Labs/stride/v16/x/autopilot/types"
+	epochtypes "github.com/Stride-Labs/stride/v16/x/epochs/types"
+	minttypes "github.com/Stride-Labs/stride/v16/x/mint/types"
+	recordstypes "github.com/Stride-Labs/stride/v16/x/records/types"
+	stakeibckeeper "github.com/Stride-Labs/stride/v16/x/stakeibc/keeper"
+	stakeibctypes "github.com/Stride-Labs/stride/v16/x/stakeibc/types"
+)
+
+func getRedeemStakeStakeibcPacketMetadata(address, ibcReceiver, transferChannel string) string {
+	return fmt.Sprintf(`
+		{
+			"autopilot": {
+				"receiver": "%[1]s",
+				"stakeibc": { "action": "RedeemStake", "ibc_receiver": "%[2]s", "transfer_channel": "%[3]s" } 
```

Comment:
the transfer channel isn't used during redemptions right? Similarly, should we rename the ibc_receiver field?

Reply:
resolved in #1030